internal/backend/user.go

```diff
@@ -102,106 +117,125 @@ func (user *user) close(ctx context.Context) error {
 
 func (user *user) deleteAllMessagesMarkedDeleted(ctx context.Context) error {
 	return user.db.Write(ctx, func(ctx context.Context, tx *ent.Tx) error {
-		ids, err := DBGetMessageIDsMarkedDeleted(ctx, tx.Client())
+		ids, err := db.GetMessageIDsMarkedDeleted(ctx, tx.Client())
 		if err != nil {
 			return err
 		}
 
-		if err := DBDeleteMessages(ctx, tx, ids...); err != nil {
+		if err := db.DeleteMessages(ctx, tx, ids...); err != nil {
 			return err
 		}
 
 		return user.store.Delete(ids...)
 	})
 }
 
-func (user *user) queueOrApplyStateUpdate(ctx context.Context, tx *ent.Tx, update stateUpdate) error {
-	// If we detect a state id in the context, it means this function call is a result of a User interaction.
-	// When that happens the update needs to be applied to the state matching the state ID immediately. If no such
-	// stateID exists or the context information is not present, all updates are queued for later execution.
-	stateID, ok := getStateIDFromContext(ctx)
-	if !ok {
-		return user.forState(func(state *State) error {
-			state.queueUpdates(update)
-			return nil
-		})
-	} else {
-		return user.forState(func(state *State) error {
-			if state.stateID != stateID {
-				state.queueUpdates(update)
-
-				return nil
-			} else {
-				if !update.filter(state) {
-					return nil
-				}
-
-				return update.apply(ctx, tx, state)
-			}
-		})
+func (user *user) queueStateUpdate(update state.Update) {
+	if err := user.forState(func(state *state.State) error {
+		state.QueueUpdates(update)
```

Comment:
Could we check the bool return value here? If it's false, the queue is closed, and it would be nice to return an error or panic (for future debugging purposes)

Reply:
The for state loop doesn't allow for anything other than error returns.

Reply:
I'll add an error log if an update couldn't be queued. The only time the queue will be closed is during shutdown.

Reply:
Done